src/server/req_runjob.c

```diff
@@ -1276,6 +1276,34 @@ parse_hook_rejectmsg(char *reject_msg, char *hook_name, int hook_name_size)
 	return NULL;
 }
 
+/**
+ * @brief
+ *		Check and put a hold on a job if it has already been run
+ *		too many times.
+ *
+ * @param[in,out]	pjob	-	job pointer
+ *
+ * @return	void
+ */
+static void
+check_failed_attempts(job *jobp)
+{
```

Comment:
Thank you for making this a function and reducing duplicate code. There is another instance of this in req_jobobit.c that also includes a NAS mod. Would be nice to consolidate that one as well.

Reply:
Good idea. I'll do.